### PR TITLE
add get to CategoricalInvRefPool

### PR DESCRIPTION
**Pull request by bkamins**

### CODE REVIEW

File: test/13_arraycommon.jl [hunk 1]
```diff
@@ -2169,10 +2169,13 @@ end
         irp = DataAPI.invrefpool(y)
         for lev in (eltype(y) >: Missing ? [missing; levels(y)] : levels(y))
             @test isequal(rp[irp[lev]], lev)
+            @test isequal(rp[get(irp, lev, nothing)], lev)
         end
 
         @test_throws KeyError irp[1]
         @test_throws KeyError irp["z"]
+        @test_throws isnothing(get(irp, 1, nothing))
+        @test_throws isnothing(get(irp, "z", nothing))
```
Comment (nalimilan) [inline test/13_arraycommon.jl:2178]:
```suggestion
        @test isnothing(get(irp, 1, nothing))
        @test isnothing(get(irp, "z", nothing))
```

File: test/13_arraycommon.jl [hunk 2]
```diff
@@ -2169,10 +2169,13 @@ end
         irp = DataAPI.invrefpool(y)
         for lev in (eltype(y) >: Missing ? [missing; levels(y)] : levels(y))
             @test isequal(rp[irp[lev]], lev)
+            @test isequal(rp[get(irp, lev, nothing)], lev)
         end
 
         @test_throws KeyError irp[1]
         @test_throws KeyError irp["z"]
+        @test isnothing(get(irp, 1, nothing))
+        @test isnothing(get(irp, "z", nothing))
```
Comment (nalimilan) [inline test/13_arraycommon.jl:2178]:
```suggestion
        @test get(irp, 1, nothing) === nothing
        @test get(irp, "z", nothing) === nothing
```

**Review by nalimilan — APPROVED** [review]
Feel free to merge and tag once the DataAPI PR has been approved and merged.

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**